### ISSUE TITLE
Validate that credentials are set before calling get, post, etc.

### ISSUE THREAD

**Issue by oma** [pusher/pusher-http-ruby#48]
This happens rarely, very rarely. it does seem sporadic. This is the ruby code
`Pusher.get("/channels/presence-oma.presentations.1", info: :user_count)[:user_count]`
It happened once before too, back then I changed the channel name, restarted the rails app, hit reload and got the same error. Right now, I've just been idle for a long time. Yeah, I know I'm not helping much... See stacktrace:

```
TypeError - no implicit conversion of nil into String:
  signature (0.1.7) lib/signature.rb:173:in `signature'
  signature (0.1.7) lib/signature.rb:53:in `sign'
  pusher (0.11.3) lib/pusher/request.rb:20:in `initialize'
  pusher (0.11.3) lib/pusher/resource.rb:29:in `create_request'
  pusher (0.11.3) lib/pusher/resource.rb:9:in `get'
  pusher (0.11.3) lib/pusher/client.rb:107:in `get'
  /Users/oma/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/forwardable.rb:171:in `get'
  app/models/presentations/presentation.rb:37:in `student_count'
```

pusher 0.11.3
ruby 2.0.0
rails 4.0.0

**Comment by ahmetabdi** [comment]
:+1:

**Comment by KevinSjoberg** [comment]
I've addressed this issue in #76. This now ensures proper configuration is in place before actually executing the given request.

**Comment by zimbatm** [comment]
Fixed in master
